src/linear_algebra/ray.lean

```diff
@@ -467,6 +467,53 @@ end
 by rw [← neg_inj, neg_neg, ← module.ray.neg_units_smul, units_smul_eq_self_iff, units.coe_neg,
   neg_pos]
 
+/-- Two vectors are in the same ray, or the first is in the same ray as the negation of the
+second, if and only if they are not linearly independent. -/
+lemma same_ray_or_same_ray_neg_iff_not_linear_independent {x y : M} :
+  (same_ray R x y ∨ same_ray R x (-y)) ↔ ¬ linear_independent R ![x, y] :=
+begin
+  by_cases hx : x = 0, { simp [hx, λ h : _root_.linear_independent R ![0, y], h.ne_zero 0 rfl] },
+  by_cases hy : y = 0, { simp [hy, λ h : _root_.linear_independent R ![x, 0], h.ne_zero 1 rfl] },
```

Comment:
Is the `_root_` needed here? If so, there's probably a `protected` missing above.

Reply:
Removed. (I'd extracted this code from a larger proof in namespace `orthonormal`, and in that namespace it was needed.)